core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -197,6 +198,16 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
 
       // Bootstrap to fetch the driver's Spark properties.
       val executorConf = new SparkConf
+
+      if (System.getenv(SecurityManager.ENV_AUTH_SECRET) != null) {
```

Comment:
The only place I see this env var set for an executor is [here](https://github.com/apache/spark/blob/7453ab0243dc04db1b586b0e5f588f9cdc9f72dd/core/src/main/scala/org/apache/spark/deploy/worker/CommandUtils.scala#L91). Shouldn't we set this at the MesosCoarseGrainedSchedulerBackend so it can be picked up here?